src/v/storage/disk_log_impl.cc

```diff
@@ -374,6 +387,13 @@ ss::future<> disk_log_impl::compact_adjacent_segments(
         }
     }
 
+    const auto sz_before = std::accumulate(
```

Comment:
nit: The 3rd parameter of the `accumulate` should be size_t, not int. Otherwise it introduces some implicit signed to unsigned int conversions inside accumulate and later in `compaction_result` c-tor.

Reply:
great hint